nltk/tag/perceptron.py

```diff
@@ -45,7 +47,14 @@ def __init__(self):
         # Number of instances seen
         self.i = 0
 
-    def predict(self, features):
+
+    def _softmax(self, scores):
+        s = np.fromiter(scores.values(), dtype=float)
+        exps = np.exp(s)
+        return exps / np.sum(exps)
```

Comment:
Understandably, numpy is used here for speed. In that case only the exponential needs to be kept. 

```python
def _softmax(self, scores):
    exps = np.exp(scores.values()) 
    return exps / np.sum(exps)
```

`np.exp()` should automatically infer the dtype to `numpy.float64`, e.g. 

```
>>> import numpy as np
>>> type(np.exp([0.1, 0.2, 0.5])[0])
<class 'numpy.float64'>
```

Reply:
scores.values() here returns a `dict_values` object which must be iterated.